account_payment_return_financial_risk/models/res_partner.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2016-2018 Tecnativa - Carlos Dauden
+# Copyright 2023 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    risk_payment_return_include = fields.Boolean(
+        string="Include Payments Returns",
+        help="Full risk computation.\n"
+        "Residual amount of move lines not reconciled with returned "
+        "lines related.",
+    )
+    risk_payment_return_limit = fields.Monetary(
+        string="Limit Payments Returns", help="Set 0 if it is not locked"
+    )
+    risk_payment_return = fields.Monetary(
+        compute="_compute_risk_account_amount",
+        compute_sudo=True,
+        string="Total Payments Returns",
+        help="Residual amount of move lines not reconciled with returned "
+        "lines related.",
+    )
+
+    @api.depends("move_line_ids.partial_reconcile_returned_ids")
+    def _compute_risk_account_amount(self):
+        self.update({"risk_payment_return": 0.0})
+        return super()._compute_risk_account_amount()
+
+    @api.model
+    def _risk_account_groups(self):
+        res = super()._risk_account_groups()
+        res["open"]["domain"] += [
+            ("partial_reconcile_returned_ids", "=", False),
+        ]
+        res["unpaid"]["domain"] += [
+            ("partial_reconcile_returned_ids", "=", False),
+        ]
+        res["returned"] = {
+            "domain": self._get_risk_company_domain()
+            + [
+                ("reconciled", "=", False),
+                ("account_type", "=", "asset_receivable"),
+                ("partial_reconcile_returned_ids", "!=", False),
+            ],
+            "fields": ["amount_residual:sum"],
+            "group_by": ["partner_id", "account_id"],
+        }
+        return res
+
+    def _prepare_risk_account_vals(self, groups):
+        vals = super()._prepare_risk_account_vals(groups)
+        vals["risk_payment_return"] = sum(
+            amount_residual
+            for (partner, account, amount_residual) in groups["returned"]["read_group"]
```

Comment:
```suggestion
            self._get_amount_in_risk_currency(
                currency, amount_residual_currency, amount_residual, account
            )
            for (partner, account, currency, amount_residual, amount_residual_currency) in groups["returned"]["read_group"]
```

Related to my previous suggestion and what is done in https://github.com/OCA/credit-control/blob/f1d0363c664d4ef94a630b31061e54c55a6153d3/account_financial_risk/models/res_partner.py#L359